### ISSUE TITLE
lol.py taking too long to execute, here is what happens after i terminate the program.

### ISSUE THREAD

**Issue by VihagChaturvedi** [nhubaotruong/league-of-legends-linux-garena-script#4]
Hello,
 Thank you so much for this guide, u are doing god's work :)
 
 I am opening this issue in reference to the step where we are supposed to run the lol.py
 
 All steps to this point have been followed religiously, here lie the errors when i terminate the program
 
 ```
 python3 lol.py
^CTraceback (most recent call last):
  File "/home/vihag/league-of-legends-linux-garena-script-main/lol.py", line 82, in <module>
    main()
  File "/home/vihag/league-of-legends-linux-garena-script-main/lol.py", line 26, in main
    riot_process = findProcessIdByName(riot_service_process_name)
  File "/home/vihag/league-of-legends-linux-garena-script-main/lol.py", line 15, in findProcessIdByName
    pinfo = proc.as_dict(attrs=["cmdline", "name", "pid"])
  File "/home/vihag/.local/lib/python3.9/site-packages/psutil/__init__.py", line 525, in as_dict
    ret = meth()
  File "/home/vihag/.local/lib/python3.9/site-packages/psutil/__init__.py", line 677, in cmdline
    return self._proc.cmdline()
  File "/home/vihag/.local/lib/python3.9/site-packages/psutil/_pslinux.py", line 1576, in wrapper
    return fun(self, *args, **kwargs)
  File "/home/vihag/.local/lib/python3.9/site-packages/psutil/_pslinux.py", line 1696, in cmdline
    with open_text("%s/%s/cmdline" % (self._procfs_path, self.pid)) as f:
  File "/home/vihag/.local/lib/python3.9/site-packages/psutil/_common.py", line 725, in open_text
    return open(fname, "rt", **kwargs)
  File "/usr/lib64/python3.9/codecs.py", line 309, in __init__
    def __init__(self, errors='strict'):
KeyboardInterrupt
```
Are the long times taken to execute the script normal? If not, what are the possible steps supposed to be taken?

Thank you so much for reading,

**Comment by nhubaotruong** [comment]
Sorry if I didn't write it clear enough in the guide, the script keeps looking for the process `RiotClientServices.exe`, so you just need to get back to the Garena client, and press `Play` then it would continue

**Comment by VihagChaturvedi** [comment]
Thank you for your reply!

> Sorry if I didn't write it clear enough in the guide, the script keeps looking for the process `RiotClientServices.exe`, so you just need to get back to the Garena client, and press `Play` then it would continue

So essentially I do this

1) Open Garena
2) Run the script
3) Then click ```play``` (Over here i suppose?)

![image](https://user-images.githubusercontent.com/45941793/128545195-cbd7e962-f814-42c8-bc37-4ae097efe3eb.png)


Here is what i get when i do so

```
--rso-auth.platform= --region=SG --locale=en_SG --priority-launch-pid=8 --priority-launch-path=C:\Riot Games\League of Legends\LeagueClient.exe --upgrade-install-path=C:\Riot Games\League of Legends --upgrade-product=league_of_legends --upgrade-patchline=live
lutris game id: 1
2021-08-06 22:10:35,775: No cores found
```

The game did run, but had terrible stability issues. And now it doesn't run at all

If possible, I am willing to contribute so that the instructions are more clear for English speakers

**Comment by nhubaotruong** [comment]
```
--rso-auth.platform= --region=SG --locale=en_SG --priority-launch-pid=8 --priority-launch-path=C:\Riot Games\League of Legends\LeagueClient.exe --upgrade-install-path=C:\Riot Games\League of Legends --upgrade-product=league_of_legends --upgrade-patchline=live
```
This is not right, it was supposed to look something like this
```
--garena-plus-token=OPkEm/84WY3kUT3JtPk2iHEYOe+8zfXTiVZbeXQ2slXf6T5VtxyGWth7dBP8pgpYwGY7ZO2Gcdo0Gus5/6dL34Mx9zmGwMvpvGkFhyRmRbBEcQFKgoe2n/D9PlKSKLOIBTLVPP/5Z9/B9O2n4c3s+ePaRIRUlI9tIpVa2+7GLdDnFqtd9wKNQcNXHVIkx6TDzjvcZ4zrFILKd11iu3hyrynj3TnelCWoqiED+qumRXVYqQtcJgggQm4moIJiG9yCYilHeOaBHm6gARpUmNG5Skx1o5fUoB/ZAGW7YxrZ0p4ymw/XLKZueZfTKPOHVTLKlDwCOm88+Vh5ziGKbTZkkg== --region=VN -- --locale=en_US --landing-token=515ac9e4557227e774ade93d0981377b9dd6bd9c812cddbdcf5940b695c2b7b2 --servers.chat.chat_host=vn1.chat.si.riotgames.com --servers.lcds.lcds_host=prodvn1.lol.garenanow.com --servers.lcds.login_queue_url=https://lqvn1.lol.garenanow.com/login-queue/rest/queues/lol --region=VN --parent-client=GarenaPC
```
Have you change the `Executable` in League of legends to `LeagueClient.exe` from the Garena-LOL that you downloaded using the Garena client ?
![image](https://user-images.githubusercontent.com/28810481/128545624-da97e6fc-5c37-4515-bdb2-31681a555023.png)

**Comment by VihagChaturvedi** [comment]
Thank you for the reply

Yes, I have changed the Lutris LOL Executable.

Here is what i did
1) Right Click on Lutris LOL
2) Configure
3)Game Options
4) Executable> Browse
5) Search for the Garena LOL Executable

Here is a before and after change

Default Directory when Lutris LOL is installed
```/home/vihag/Games/league-of-legends/drive_c/Riot Games/League of Legends/LeagueClient.exe```

Here is what the Directory looks like when I changed the Executable of Lutris LOL -> Garena LOL (i essentially replaced the default location of Riot LOL to Garena LOL)

```/home/vihag/Games/garena/drive_c/Garena/Games/32771/LeagueClient/LeagueClient.exe```

After doing this, I somehow get both versions launched simultaneously(see image below)

There is no internet access on the second one for some reason.

FYI, I have BOTH Lutris LOL and Garena LOL installed, I have only changed the default path of executable of the Lutris LOL to that of Garena LOL


![image](https://user-images.githubusercontent.com/45941793/128549682-4abc5d3c-9917-4f7d-b662-268eb509de8b.png)

**Comment by nhubaotruong** [comment]
Try close all the game and Garena and run `wineserver -k` in terminal to kill all wine processes.

Then check `System Monitor` application of your DE and check if any `League` and `Riot` related process is running, kill all of them

Then try again.

And you don't need to be so formal, casual chat is ok with me :)

**Comment by VihagChaturvedi** [comment]
Heh, I tend to be formal as it is often times the best way to show someone respect for their work, thank you for the reassurance :)

So here are the steps to what I should do.
1) Launch Garena Client
2) Apply ```lol.py``` patch
3) Then click ```Play```

I only get results after I do this.

Is it necessary to run the script every time I wish to play Garena LOL?

**Comment by nhubaotruong** [comment]
Yes you will have to run the script every time you want to play LOL, the Garena token only lasted for about 5 minutes

**Comment by VihagChaturvedi** [comment]
I see, that's a _bit_ inconvenient, but its alr, better than playing on 200+ Ping in EUW servers.

I can see why this tutorial can be a bit confusing. Maybe a decent idea to re-write the README.md or atleast rectify a few things I found a bit confusing

I would like to contribute to the README.md to make sure no one has the same issues as I did, along with perhaps a FAQ page.

But I need to properly learn how to use GitHub for that :P

Anyways, imma see what I can do. Good luck and have a great day!
